Update placeholder, create placeholder example

smfoote opened the pull request:
Placeholder now uses Fiber and the state decorator. The example (examples/placeholder.html) includes both input[type=text] and textarea, inside a valid for with a working submit button.

Commit:
Update placeholder, create placeholder example

Placeholder now uses Fiber and the state decorator. The example (examples/placeholder.html) includes both input[type=text] and textarea, inside a valid for with a working submit button.

Commit:
Fix title in placeholder.html - add some styling

Iheartweb commented on scripts/components/Placeholder.js:
```diff
@@ -30,29 +33,30 @@ Placeholder = FormElement.extend( function ( base ) {
+        if ( Placeholder.hasState( PLACEHOLDER_STATE ) ) {
           $element.val('');
-          $element.removeClass( PLACEHOLDER_CLASS );
+          Placeholder.clear();
```
Use an explicit state removal here `Placeholder.removeState( PLACEHOLDER_STATE )`

Iheartweb commented on scripts/components/Placeholder.js:
```diff
@@ -6,6 +6,9 @@
 var FormElement = require( 'lu/FormElement' ),
+    helpers = require( 'lu/helpers' ),
+    StateDecorator = require( 'lu/decorators/state' ),
+    Fiber = require( 'Fiber' ),
```
require constants.

Iheartweb commented on examples/placeholder.html:
```diff
@@ -0,0 +1,72 @@
+
+    window.LU_DEBUG = 5;
+
+    Lu.map( $( 'form' ).find( 'input[type=text], textarea' ), 'Placeholder', function( $element, component ){} );
```
move this mapper into mappers.js Also, after my pull request is through, `component` will not be passed as a param. Instead `this === component`

Iheartweb replied:
also this should limit to inputs/textareas with a placeholder and data-lu="Placeholder"

Iheartweb commented on scripts/components/Placeholder.js:
```diff
@@ -30,29 +33,30 @@ Placeholder = FormElement.extend( function ( base ) {
 
       _.defaults( settings, defaults );
       base.init.call( Placeholder, $element, settings );
-      PLACEHOLDER_CLASS = settings.placeholderClass;
```
I am not sure this is needed. Perhaps we can just have a hard coded state in constants.

Iheartweb commented:
Nice. Lint it. Add it to the build. See the readme on my branch.... grunt is pending a pull request.

Commit:
Add Placeholder to mappers.js

Commit:
Merge branch 'base'

Commit:
Resolve merge conflict

Commit:
Add tests for Placeholder, update mappers.js

New tests for Placeholder and update mappers.js to include placeholder.

Commit:
Update Placeholder to use constants.

Also, update constants to include a PLACEHOLDER constant. Finally, in Placeholder.js, explicitly remove the placeholder state, instead of using clear.

Commit:
Update grunt build script to include Placeholder

This includes unit tests and build.

Commit:
Fix merge conflicts

Commit:
Lint cleanup and adding myself to the contributor list

Iheartweb commented:
@smfoote  This PR cannot be auto merged. Do a git pull linkedin master and commit the merge to your fork. Thanks!

Commit:
Fix merge conflicts in package.json and grunt.js
